### PR TITLE
Improve user info test cases

### PR DESCRIPTION
**Pull request by Thumimku**
Public Issue : https://github.com/wso2/product-is/issues/21313

Current test user admin for authorization which is unnecessary, we should use business user.
Current suite lacks some negative cases which is using Invalid token access to get user info.
Apart from this fix will introduce test cases for
- POST request
- validate user info for M2M token

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OpenIdUserInfoTestCase.java [hunk 1]
```diff
@@ -336,6 +334,120 @@ public void testValidateTokenScope() throws Exception {
...
+        JSONParser parser = new JSONParser();
+        JSONObject jsonResponse = (JSONObject) parser.parse(responseString);
+        if (jsonResponse == null) {
+            throw new Exception("Error occurred while getting the response.");
+        }
+        Assert.assertNotNull(jsonResponse.get("sub"), "sub from introspection endpoint response is null.");
+        Assert.assertNotNull(jsonResponse.get("email"), "sub from introspection endpoint response is null.");
+    }
+
+    @Test(groups = "wso2.is", description = "request user info using POST iwht invalid token",
```
Comment (shashimalcse) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OpenIdUserInfoTestCase.java:364]:
```suggestion
    @Test(groups = "wso2.is", description = "request user info using POST with invalid token",
```

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OpenIdUserInfoTestCase.java [hunk 2]
```diff
@@ -336,6 +334,120 @@ public void testValidateTokenScope() throws Exception {
...
+        if (jsonResponse == null) {
+            throw new Exception("Error occurred while getting the response.");
+        }
+        Assert.assertEquals(jsonResponse.get("error_description"),
+                "Access token validation failed", "Unexpected error description");
+        Assert.assertEquals(jsonResponse.get("error"), "invalid_token",
+                "Unexpected error message");
+    }
+
+    @Test(groups = "wso2.is", description = "Send authorize user request without openid and invoke introspect",
```
Comment (shashimalcse) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OpenIdUserInfoTestCase.java:394]: is this description correct. we are sending openid specc but its m2m

Comment (Thumimku) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OpenIdUserInfoTestCase.java]: fixed with https://github.com/wso2/product-is/pull/21361/commits/be2f26ad6ce18d03ba9390625c6d1d6ab2962dcd

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OpenIdUserInfoTestCase.java (hunk 1 above)
Comment (Thumimku) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OpenIdUserInfoTestCase.java]: fixed with https://github.com/wso2/product-is/pull/21361/commits/be2f26ad6ce18d03ba9390625c6d1d6ab2962dcd

**Review by shashimalcse — APPROVED** [review]

**Pull request merged**